mongofiles/mongofiles_test.go

```diff
@@ -476,67 +484,124 @@ func TestMongoFilesCommands(t *testing.T) {
 			})
 		})
 
-		Convey("Testing the 'put' command by putting some lorem ipsum file with 287613 bytes should", func() {
-			filename := "lorem_ipsum_287613_bytes.txt"
-			mf, err := simpleMongoFilesInstanceWithFilename("put", filename)
+		Convey("Testing the 'put' command with copies of a lorem ipsum file with 287613 bytes should", func() {
+			const (
+				numCopies  = 3
+				numBytes   = 287613
+				copyFormat = "lorem_ipsum_copy_%v.txt"
+			)
+
+			originalFile := util.ToUniversalPath("testdata/lorem_ipsum_287613_bytes.txt")
+
+			testFiles := make([]string, numCopies)
+			for i := 0; i < numCopies; i++ {
+				newFile := util.ToUniversalPath("testdata/" + fmt.Sprintf(copyFormat, i))
+
+				// Makes new copies of lorem ipsum file
+				err = func(src, dst string) error {
+					in, err := os.Open(src)
+					if err != nil {
+						return err
+					}
+					defer in.Close()
+
+					out, err := os.Create(dst)
+					if err != nil {
+						return err
+					}
+					defer out.Close()
+
+					_, err = io.Copy(out, in)
+					if err != nil {
+						return err
+					}
+
+					return nil
+				}(originalFile, newFile)
+				So(err, ShouldBeNil)
+
+				testFiles[i] = newFile
+			}
+
+			mf, err := simpleMongoFilesInstanceWithMultipleFileNames("put", testFiles...)
 			So(err, ShouldBeNil)
-			So(mf, ShouldNotBeNil)
-			mf.StorageOptions.LocalFileName = util.ToUniversalPath("testdata/" + filename)
 
 			var buff bytes.Buffer
 			log.SetWriter(&buff)
 
-			Convey("insert the file by creating two chunks (ceil(287,613 / 255 * 1024)) in GridFS", func() {
-				buff.Truncate(0)
-				str, err := mf.Run(false)
+			str, err := mf.Run(false)
+			So(err, ShouldBeNil)
+			So(str, ShouldBeEmpty)
+
+			Convey("log an event specifying the completion of each file", func() {
+				const (
+					logAdding = "adding gridFile: %v"
+					logAdded  = "added gridFile: %v"
+				)
+
+				logOutput := buff.String()
+
+				for _, testFile := range testFiles {
+					So(logOutput, ShouldContainSubstring, fmt.Sprintf(logAdding, testFile))
+					So(logOutput, ShouldContainSubstring, fmt.Sprintf(logAdded, testFile))
+
+				}
+			})
+
+			Convey("and files should exist in GridFS", func() {
+				bytesGotten, err := getFilesAndBytesListFromGridFS()
 				So(err, ShouldBeNil)
-				So(str, ShouldEqual, "")
-				So(buff.Len(), ShouldNotEqual, 0)
 
-				Convey("and files should exist in gridfs", func() {
-					bytesGotten, err := getFilesAndBytesListFromGridFS()
-					So(err, ShouldBeNil)
-					So(len(bytesGotten), ShouldEqual, len(testFiles)+1)
```

Comment:
I don't understand what you mean, since I check for the number of bytes gotten in lines 556-557. Could you be more specific about what exactly I'm missing?

Reply:
Check number of files in `bytesGotten` -- GridFS should contain the test files and _only_ the test files.

Reply:
I just thought a way to improve the current test, for `get` test, instead of trying to output all the files written to db, maybe you can write 4 files and have 3 filenames in the list?